migrations/versions/9bee3b519bf1_add_history_fingerprint.py

```diff
@@ -0,0 +1,57 @@
+"""add_history_fingerprint
+
+Revision ID: 9bee3b519bf1
+Revises: 167a97c81739
+Create Date: 2023-08-29 19:33:34.654753
+
+"""
+from datetime import datetime
+
+import sqlalchemy as sa
+from alembic import op
+
+# revision identifiers, used by Alembic.
+revision = "9bee3b519bf1"
+down_revision = "167a97c81739"
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    print(f"{datetime.now()} adding column")
+    op.add_column(
+        "benchmark_result", sa.Column("history_fingerprint", sa.Text(), nullable=True)
+    )
+
+    print(f"{datetime.now()} backfilling")
+    op.execute(
+        """
+        UPDATE benchmark_result SET history_fingerprint = MD5(
+            benchmark_result.case_id ||
+            benchmark_result.context_id ||
+            hardware.hash ||
+            benchmark_result.commit_repo_url
+        )
```

Comment:
Oh, did we confirm that this method yields the exact same output as https://github.com/conbench/conbench/pull/1463/files#diff-9d75d1bf6ac23bfc056a410721cf4c9bccd8b302f8e3e83e4b4bf1698d0eaad7R1044?

It's a bit of a risky move because yes the input to a hash function is a raw byte sequence and here we let the database use some of its implementation detail for how to store text, I think. This can only work (result in the same hash) if in both cases we use UTF-8-encoded text, and maybe that's true.

Reply:
Yes, I should have added a code comment. I agree that this is risky but I validated that it works for at least one real-world history fingerprint.